### DISCUSSION TITLE
Print failure near the end of print

### DISCUSSION THREAD

**Discussion by tjhayko** [bassamanator/Sovol-SV06-firmware#187] [Q&A]
I've had a print failure near the end of reasonably small print, but I couldn't get klipper on my BTT Pad 7 to respond after that, and I had to reboot it.

I should have gotten a screenshot, but the message was something like "mcu connection lost".  

Which logs should I try to use to troubleshoot this?

Thanks in advance!

**Comment by bassamanator** [comment]
`~/printer_data/logs/klippy.log` and `~/printer_data/logs/moonraker.log` are the usual places to look.

Sounds like a bad usb connection or mcu crash.

**Comment by tjhayko** [comment]
Thanks.  Not much of anything showing in those logs, so I'm trying again.

**Comment by sydniiiiiii** [comment]
For troubleshooting the "MCU connection lost" error in Klipper on your BTT Pad 7, follow these steps:

Check Klipper Logs:

Access the klippy.log file, which logs all Klipper-related activity. This file is generally located in the same directory where Klipper is installed. It will contain details of what led to the error and might show connection issues or command errors.
Look for lines around the time the print failed or the connection was lost. These may give specific error codes or messages indicating why the MCU connection failed.
Examine Octoprint Logs (if applicable):

If you’re using Octoprint with Klipper, check the octoprint.log for any communication or connectivity errors around the same timestamp as the failure.
Review the MCU Connection:

Sometimes the "MCU connection lost" error occurs due to loose cables or a temporary power fluctuation. Check all connections between the BTT Pad 7 and your printer’s mainboard to ensure they’re secure.
Enable Debug Logging:

If the cause isn’t clear, consider enabling debug-level logging in Klipper temporarily to capture more detail on the issue. This may help isolate hardware or configuration issues.
Once you review these logs and steps, you should have more information to pinpoint the root cause of the failure.
